include/libtorrent/alert_types.hpp

```diff
@@ -274,12 +274,14 @@ TORRENT_VERSION_NAMESPACE_2
 	{
 		// internal
 		TORRENT_UNEXPORT torrent_removed_alert(aux::stack_allocator& alloc
-			, torrent_handle const& h, info_hash_t const& ih);
+			, torrent_handle const& h, info_hash_t const& ih, void* clientdata);
 
 		TORRENT_DEFINE_ALERT_PRIO(torrent_removed_alert, 4, alert_priority::critical)
 		static constexpr alert_category_t static_category = alert::status_notification;
 		std::string message() const override;
+
 		info_hash_t info_hash;
+		void* clientdata;	// largely redundant with info_hash
```

Comment:
please put the comment above the `clientdata` field, and make it actually explain what it is and where it comes from. This is used to generate the documentation.